en/mavgen_c/defines.md

```diff
@@ -0,0 +1,31 @@
+# MAVLink C Library #defines
+
+The following C `#defines` can be set in code in order to tune the setup for use on different platforms.
+
+- `MAVLINK_USE_CONVENIENCE_FUNCTIONS` ([protocol.h](https://github.com/ArduPilot/pymavlink/blob/master/generator/C/include_v2.0/protocol.h)): Causes convenience functions to be defined, including: `_mav_finalize_message_chan_send()`, `_mavlink_send_uart()`, `_mavlink_resend_uart()`. To use, add `#define MAVLINK_USE_CONVENIENCE_FUNCTIONS` to your code.
```

Comment:
Why wouldn't you do this? Would there be any real saving in memory?

Reply:
This just gives you a way of implementing these functions yourself. For instance in PX4 we need to send broadcast messages on all MAVLink instances, not just one, so the function is redefined to go through all instances and send off the data, see:
https://github.com/PX4/PX4-Autopilot/blob/6f861ba8897c67592627d4fe3011a4e19bef7942/src/modules/mavlink/mavlink_main.cpp#L88.

Reply:
Thanks @julianoes - I don't get that from the example - the linked example still takes a channel so sends on a particular instance rather than going through all of them. The equivalent convenience function does something similar https://github.com/ArduPilot/pymavlink/blob/15fc4778844d94ccf83d5f909cf45d07ac9dbfe4/generator/C/include_v2.0/mavlink_helpers.h#L1130

Could I get away with something like:

```suggestion
- `MAVLINK_USE_CONVENIENCE_FUNCTIONS` ([protocol.h](https://github.com/ArduPilot/pymavlink/blob/master/generator/C/include_v2.0/protocol.h)): Causes convenience functions to be defined, including: `_mav_finalize_message_chan_send()`, `_mavlink_send_uart()`, `_mavlink_resend_uart()`.
  Library users may choose not to set this and roll out their own implementations for efficiency (or other) reasons.
```

Reply:
Ok.